.github/workflows/check-circular-deps.yml

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+name: Circular dependency check
+
+on:
+  push:
+    branches: [3.0*]
+  pull_request:
+    branches: [3.0*]
+
+jobs:
+  spec-check:
+    name: Circular dependency check
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout the branch of our repo that triggered this action
+      - name: Workflow trigger checkout
+        uses: actions/checkout@v4
+
+      - name: Check for circular dependencies
+        run: |
+          echo "Checking for circular dependency loops..."
+          pushd toolkit
+          ./scripts/setuplkgtoolchain.sh
+          BUILD_ID=$(wget -qO - https://mariner3dailydevrepo.blob.core.windows.net/lkg/lkg-3.0-dev.json | jq -r ".dailybuildid" | tr '\.' '-')
+          sudo make -j$(nproc) graph REBUILD_TOOLS=y DAILY_BUILD_ID=${BUILD_ID}
```

Comment:
Could we only add a comment to explain these steps?

Reply:
Sure, updated